Tests/SentryTests/Categories/NSMutableDictionarySentryTests.swift

```diff
@@ -0,0 +1,54 @@
+import XCTest
+
+class NSMutableDictionarySentryTests: XCTestCase {
+
+    func testEmptyDictionary() {
+        let empty = NSMutableDictionary()
+        
+        empty.mergeEntries(from: [String: String]())
+        
+        XCTAssertEqual(empty, NSMutableDictionary())
+    }
+    
+    func testEmptyDict_AddsDict() {
+        let dict = NSMutableDictionary()
+        dict.mergeEntries(from: [0: [1: 1]])
+        
+        XCTAssertEqual([0: [1: 1]], dict as? [Int: [Int: Int]])
+    }
+    
+    func testTwoEntries_ExistingGetsMerged() {
+        let dict = NSMutableDictionary(dictionary: [0: [0: 0], 1: [0: 0]])
+        dict.mergeEntries(from: [0: [1: 1]])
+        
+        XCTAssertEqual([0: [0: 0, 1: 1], 1: [0: 0]], dict as? [Int: [Int: Int]])
+    }
+    
+    func testTwoEntries_SameGetsOverwritten() {
+        let dict = NSMutableDictionary(dictionary: [0: [0: 0]])
+        dict.mergeEntries(from: [0: [0: 1]])
+        
+        XCTAssertEqual([0: [0: 1]], dict as? [Int: [Int: Int]])
+    }
+    
+    func testTwoLevelsNested_ExistingGetsMerged() {
+        let dict = NSMutableDictionary(dictionary: [0: [0: [0: 0]]])
+        dict.mergeEntries(from: [0: [0: [1: 1]]])
+        
+        XCTAssertEqual([0: [0: [0: 0, 1: 1]]], dict as? [Int: [Int: [Int: Int]]])
+    }
+    
+    func testExistingNotADict_NewIsADict_GetsOverwritten() {
+        let dict = NSMutableDictionary(dictionary: [0: 0])
+        dict.mergeEntries(from: [0: [1: 1]])
+        
+        XCTAssertEqual([0: [1: 1]], dict as? [Int: [Int: Int]])
+    }
+    
+    func testExistingIsADict_NewIsNotADict_GetsOverwritten() {
+        let dict = NSMutableDictionary(dictionary: [0: [0: 0]])
+        dict.mergeEntries(from: [0: 1])
+        
+        XCTAssertEqual([0: 1], dict as? [Int: Int])
+    }
```

Comment:
What are the real-world equivalents of these? Seems like it'd lead to some kind of data loss. Or is it upgrading the type of something? Seems like we'd more want to a merge produce an array of values for a particular key if it already exists, but that leads to an issue of heterogenous types in that array. Do we have such heterogeneity anywhere in practice?

Reply:
I think @philipphofmann meant to make a more generic test, objc Dictionary it is not strongly typed and anything can happen. 
Since the code this is testing is an auxiliary non specific code, it's worth testing this behaviour.

Reply:
It's being used to merge contexts here: https://github.com/getsentry/sentry-cocoa/pull/1969/files#diff-8ba9297eda94f834f1fc0767e1550efd533b9a402b79f85bf829f64a0f04b7b0L515-L521 

If this can drop data from someone's context, I think it would be a better approach to at least check the types inside the dictionary during the merge and print a warning to let people know they're going to lose data in the merge due to heterogenous types.

The reason I asked about real-world equivalents is because I'd think ideally we'd move away from dictionaries altogether, especially if its something that's going to be parsed by our backend. But if this is something that only customers will ever write to and read from, then yes, they should have the freedom to do what they want.

Reply:
We use dictionaries for the context, so users can put in their custom data if they want to. Dictionaries give them the advantage of using deep nesting for their data. We add the app start metrics to that context. The problem is that we need to update a nested value inside that dictionary. I just got tired of writing complex code of getting the dictionary inside a key of the context dictionary, making it mutable, adding a new value, and then adding it again to the dictionary. Therefore, I added `mergeEntries` to make this process easier.

We could start a discussion if we want to switch to a typed context, which still allows you to add your custom nested data, but this would be a breaking change, and we have to wait until the next major. 

Most Sentry APIs are designed for languages where dictionaries are standard, like JavaScript and Python. Sometimes we need to change them and adapt them to strongly typed languages, Swift and ObjC.